### ISSUE TITLE
The check for did_load_filetypes is too strict

### ISSUE THREAD

**Issue by lervag** [junegunn/vim-plug#1175]
Neovim v0.7 introduces the opt-in `filetype.lua` to replace `filetype.vim` possibly in v0.8. To use this and disable `filetype.vim`, one is suggested to add

```vim
let g:do_filetype_lua = 1
let g:did_load_filetypes = 0
```

to ones `init.vim` file. If you add this before `plug#end()` is called, then vim-plug will unset `g:did_load_filetypes` due to this code:

https://github.com/junegunn/vim-plug/blob/93ab5909784e09134e90f15cafa8a5edcc9a00fe/plug.vim#L355-L357

I suggest changing the conditional to this:

```vim
if exists('g:did_load_filetypes') && g:did_load_filetypes > 0
  filetype off
endif
```

See e.g. [@gpanders's blog post](https://gpanders.com/blog/whats-new-in-neovim-0-7/#filetypelua) for some more writeup on this particular feature in neovim.

**Comment by Shougo** [comment]
I think this is better.

`if get('g:', 'did_load_filetypes', 0)`

**Comment by junegunn** [comment]
Thanks. Would you be willing to submit a pull request?

**Comment by lervag** [comment]
Yes, sure. I'm closing this as it is addressed in #1176.
